app/routes/campaigns.js

```diff
@@ -90,37 +100,62 @@ router.post('/:id/message', (req, res) => {
     return helpers.sendResponse(res, 422, 'Missing required message type parameter.');
   }
 
-  // Check that campaign is active.
-  // TODO: Refactor this to no longer use app.locals.campaigns -- would need to query Phoenix API
-  // to check for Campaign Status.
-  const campaign = app.locals.campaigns[campaignId];
-  if (!campaign || campaign.status !== 'active') {
+  if (!helpers.isCampaignBotCampaign(campaignId)) {
     const msg = `Campaign ${campaignId} is not running on CampaignBot`;
     return helpers.sendResponse(res, 422, msg);
   }
 
-  // Check that campaign suports requested message type.
-  const messageBody = campaign.messages[type];
-  if (!messageBody) {
-    const msg = `Campaign ${campaignId} does not support '${type}' messages`;
-    return helpers.sendResponse(res, 422, msg);
-  }
+  let messageBody;
 
-  return app.locals.clients.northstar.Users.get('mobile', phone)
-  .then((user) => {
-    mobilecommons.send_message(phone, messageBody);
-    const msg = `Sent text for ${campaignId} ${type} to ${user.mobile}`;
-    logger.info(msg);
-    stathat('Sent campaign message');
-    return helpers.sendResponse(res, 200, msg);
-  })
-  .catch((err) => {
-    if (err.response) {
-      logger.error(err.response.error);
-    }
-    const msg = `Error sending text to user #${phone}: ${err.message}`;
-    return helpers.sendResponse(res, 500, msg);
-  });
+  return app.locals.db.campaigns.findById(campaignId)
+    .then((campaignDoc) => {
+      logger.debug(`found campaign:${campaignId}`);
+
+      // Check that campaign suports requested message type.
+      messageBody = campaignDoc.messages[type];
+      if (!messageBody) {
+        const msg = `Campaign ${campaignId} does not support '${type}' messages.`;
+        logger.error(msg);
+        const err = new UnprocessibleEntityError(msg);
+        throw err;
+      }
+
+      return phoenix.Campaigns.get(campaignId);
+    })
+    .then((phoenixCampaign) => {
+      logger.debug(`phoenix.Campaigns.get found campaign:${campaignId}`);
+      if (phoenixCampaign.status === 'closed') {
+        const msg = `Campaign ${campaignId} is closed.`;
+        throw new UnprocessibleEntityError(msg);
+      }
+
+      return northstar.Users.get('mobile', phone);
+    })
+    .then((user) => {
+      logger.debug(`northstar.Users.get found user:${user.id}`);
+
+      mobilecommons.send_message(phone, messageBody);
+      const msg = `Sent text for ${campaignId} ${type} to ${user.mobile}`;
+      logger.info(msg);
+      stathat('Sent campaign message');
+
+      return helpers.sendResponse(res, 200, msg);
+    })
+    // TODO: Catch UnprocessibleEntityError. Commenting this code out because it is
+    // throwing a "Class constructors cannot be invoked without 'new'"" error
+    // .catch(UnprocessibleEntityError, (err) => {
```

Comment:
This is the weirdest thing. I'm copying https://github.com/DoSomething/gambit/blob/3ca61cfc71be5c54e871ee271cfd67b7a1dfa8b1/app/routes/chatbot.js#L304 which doesn't throw this error.